docusaurus/docs/ui-extensions/create-an-extension-point.md

```diff
@@ -15,40 +15,288 @@ keywords:
+- If the extension-registry changes
+
+Be sure to only change the `context` object if its content changes; otherwise, you could create unnecessary re-renders. The following example shows how to approach these scenarios in a safe way:
+
+```tsx
+import { useMemo } from 'react';
+import { usePluginLinkExtensions } from '@grafana/runtime';
+
+function AppMenuExtensionPoint({ referenceId }) {
+  // Instead of defining the object here (which would result in a new object on every render),
+  // we use `useMemo()` to only update the context object when its "dynamic" dependencies change.
+  const context = useMemo(
+    () => ({
+      referenceId,
+      timeZone: 'UTC',
+    }),
+    [referenceId]
+  );
+  const { extensions } = usePluginLinkExtensions({
+    extensionPointId: 'plugins/another-app-plugin/menu',
+    context,
+  });
+
+  if (extensions.length === 0) {
+    return null;
+  }
+
+  return (
+    <div>
+      {extensions.map((extension) => {
+        return (
+          <a href={extension.path} onClick={extension.onClick} title={extension.description} key={extension.key}>
+            {extension.title}
+          </a>
+        );
+      })}
+    </div>
+  );
+}
+```
+
+#### Example - rendering component extensions
+
+:::note
+
+\*\*This feature is only available in Grafana versions 10.1.0 and above.
+
+:::
+
+Component type extensions are simple React components, which gives you much more freedom in what you can make them do. You can pass contextual information to the extension components using props.
+
+```tsx title="src/components/Toolbar.tsx"
+import { usePluginComponentExtensions } from '@grafana/runtime';
+
+export const Toolbar = () => {
+  // This only returns type="component" extensions
+  // Heads up! We don't specify a context object below, we pass in the contextual information as a prop to the component later.
+  const { extensions } = usePluginComponentExtensions({ extensionPointId: '<extension-point-id>' });
 
-In the following example, a `<LinkButton />`-component is rendered for all link extensions that other plugins registered for the `plugins/another-app-plugin/menu` extension point ID. The context is passed as the second parameter to `getPluginLinkExtensions`, which makes the context immutable before passing it to other plugins.
+  return (
+    <div>
+      <div className="title">Title</div>
+      <div className="extensions">
+        {/* Loop through the available extensions */}
+        {extensions.map((extension) => {
+          const Component = extension.component as React.ComponentType<{
+            version: string;
+          }>;
+
+          // Render extension component and pass contextual information (version)
+          return (
+            <div key={extension.id}>
+              <Component version="1.0.0" />
+            </div>
+          );
+        })}
+      </div>
+    </div>
+  );
+};
+```
+
+### The `getPluginExtensions()` method - _deprecated_
```

Comment:
Do we want to keep this in the docs?

Reply:
I think until it is actually removed, we should?

Reply:
The migration process is just to swap the hook right? (Which we could mention). 

I worry it's just noise that makes the doc look longer and more complicated. Especially given we don't want to encourage any further adoption, and we believe there's currently limited community uptake. 

I guess we need it until the core extension points are all converted. But after that I would personally just remove it.

Reply:
> The migration process is just to swap the hook right? (Which we could mention).

Basically yes - however if they have tests then updating those is bit more work (due to mocking).


> I worry it's just noise that makes the doc look longer and more complicated.

I agree.


Yeah, let's keep them for now (in this PR) and let's chat about when we should remove them. My other concern is that we might don't even want to remove them / deprecate them in the end, but clearly communicate what they can be used for. (We need to think about the use cases, but a function like this could be use in places outside of a React component, which can come handy. It needs some more thought.)